packages/core/http/core-http-server-internal/src/csp/csp_directives.test.ts

```diff
@@ -121,5 +121,28 @@ describe('CspDirectives', () => {
         `"script-src 'report-sample' 'self' 'unsafe-hashes'; worker-src 'report-sample' 'self' blob:; style-src 'report-sample' 'self' 'unsafe-inline'"`
       );
     });
+
+    it('merges additional CSP configs as expected', () => {
+      const config = cspConfig.schema.validate({
+        connect_src: ['*.foo.bar'], // should de-dupe these
+      });
+      const additionalConfig1 = {
+        connect_src: ['*.foo.bar'],
+        img_src: ['*.foo.bar'],
+      };
+      const additionalConfig2 = {
+        connect_src: [`cdn.host.test`],
+        font_src: [`cdn.host.test`],
+        frame_src: [`cdn.host.test`],
+        img_src: [`cdn.host.test`],
+        worker_src: [`cdn.host.test`],
+        script_src: [`cdn.host.test`],
+        style_src: [`cdn.host.test`],
+      };
+      const directives = CspDirectives.fromConfig(config, additionalConfig1, additionalConfig2);
+      expect(directives.getCspHeader()).toEqual(
+        `script-src 'report-sample' 'self' cdn.host.test; worker-src 'report-sample' 'self' blob: cdn.host.test; style-src 'report-sample' 'self' 'unsafe-inline' cdn.host.test; connect-src 'self' *.foo.bar cdn.host.test; font-src 'self' cdn.host.test; frame-src 'self' cdn.host.test; img-src 'self' *.foo.bar cdn.host.test`
```

Comment:
Tested this [here](https://cspvalidator.org/#headerValue%5B%5D=script-src+'report-sample'+'self'+cdn.host.test%3B+worker-src+'report-sample'+'self'+blob%3A+cdn.host.test%3B+style-src+'report-sample'+'self'+'unsafe-inline'+cdn.host.test%3B+connect-src+'self'+*.foo.bar+cdn.host.test%3B+font-src+'self'+cdn.host.test%3B+frame-src+'self'+cdn.host.test%3B+img-src+'self'+*.foo.bar+cdn.host.test&strategy=intersection)